libweasyl/libweasyl/staff.py

```diff
@@ -1,70 +1,35 @@
 """
-Dictionaries of Weasyl staff.
-
-This module must be initialized with a call to `staff._init_staff()`
-with the path to a yaml config file where staff members are identified
-by their user id. Example contents of such a config file might look as
-follows:
-
-    directors:
-        - 1014   # Fiz
-        - 2061   # Ikani
-
-    technical_staff:
-        - 5756   # Weykent
-
-    admins:  # Directors and technical_staff also have admin privs.
-        - 23613  # Hendikins
-        - 3      # Kihari
-
-    mods:  # Admins also have mod privs.
-        - 40212  # pinardilla
-
-    developers:
-        - 38623  # 8BitFur
-        - 34165  # Charmander
-        - 15224  # Foximile
-        - 2475   # Kailys
-        - 8627   # Kauko
-
-We recommend storing staff members in lexicographic order by name
-within each group.
+Sets of Weasyl staff user ids.
 """
 
-import yaml
-
-
-DIRECTORS = set()
-""" Directors have the same powers as admins. """
-
-
-TECHNICAL = set()
-""" Technical staff can moderate all content and manage all users. """
-
-
-ADMINS = set()
-""" Site administrators can update site news and moderate user content. """
-
+def _init_staff(directors, technical_staff, admins, mods, developers):
+    """
+    Populates staff members from passed kwargs.
 
-MODS = set()
-""" Site moderators can hide submissions, manage non-admin users, etc. """
+    Parameters:
+        directors: Array with directors
+        technical_staff: Array with technical staff
+        admins: array with admins
+        mods: Array with mods
+        developers: Array with developers
+    """
 
+    """ Directors have the same powers as admins. """
+    global DIRECTORS
+    DIRECTORS = frozenset(directors)
 
-DEVELOPERS = set()
-""" Purely cosmetic group for users who contribute to site development. """
+    """ Technical staff can moderate all content and manage all users. """
+    global TECHNICAL
+    TECHNICAL = DIRECTORS | frozenset(technical_staff)
 
+    """ Site administrators can update site news and moderate user content. """
+    global ADMINS
+    ADMINS = TECHNICAL | frozenset(admins)
 
-def _init_staff(staff_config_path):
-    """
-    Loads staff from a yaml config file.
+    """ Site moderators can hide submissions, manage non-admin users, etc. """
+    global MODS
+    MODS = ADMINS | frozenset(mods)
 
-    Parameters:
-        staff_config_path: String path to a yaml file specifying staff.
-    """
-    with open(staff_config_path) as config_file:
-        staff = yaml.safe_load(config_file)
-    DIRECTORS.update(staff['directors'])
-    TECHNICAL.update(DIRECTORS, staff['technical_staff'])
-    ADMINS.update(DIRECTORS, TECHNICAL, staff['admins'])
-    MODS.update(ADMINS, staff['mods'])
-    DEVELOPERS.update(staff['developers'])
+    """ Purely cosmetic group for users who contribute to site development. """
+    global DEVELOPERS
+    DEVELOPERS = frozenset(developers)
```

Comment:
The docstrings here aren't going to be used for anything. Since there were on module-level assignments before, sphinx would pick them up as documentation for the module attributes exposed that way. The best way to handle this, I think, is to keep the top-level assignments like:

```
DIRECTORS = frozenset()
""" Directors have the same powers as admins. """
```

and then replace them from `_init_staff`:

```
    global DEVELOPERS
    DEVELOPERS = frozenset(developers)
```

just, with the docstrings omitted from the function body.